### PR TITLE
[community:r] add BiocCheck to r.rb in travis-build

### PR DESCRIPTION
**Pull request by LiNk-NY**
Hi Jim and Craig, @jimhester @craigcitro 

Here is my attempt at the pull request mentioned in travis-ci/travis-ci#9455

Users that add: 
```
r: bioc-devel

bioc_check: true
```
Should get a check of the package with `R CMD BiocCheck`.

I've tried to test this locally but I am unable to build the `ci.sh` from the `travis compile` command (I'm using the `travisci/ci-garnet:packer-1512502276-986baf0`. I suspect that I have to install additional dependencies. Any pointers are appreciated. I'm using this as reference: https://stackoverflow.com/a/45761267

Regards,
Marcel

### CODE REVIEW

File: lib/travis/build/script/r.rb [hunk 1]
```diff
@@ -236,6 +237,10 @@ def script
               "#{config[:r_check_args]}"
             sh.cmd "R CMD check \"${PKG_TARBALL}\" #{config[:r_check_args]}; "\
               "CHECK_RET=$?", assert: false
+            if config[:bioc_check]
```
Comment (jimhester) [inline lib/travis/build/script/r.rb]: I think maybe this should be in a separate fold

**Comment by LiNk-NY** [comment]
Hi Jim, @jimhester 
Thanks for taking a look at it!
I have made the changes you suggested. 

Best regards,
Marcel

File: spec/build/script/r_spec.rb [hunk 2]
```diff
@@ -192,6 +192,11 @@
                            assert: true, echo: true, retry: true, timing: true]
     end
 
+    it 'does BiocCheck if requested' do
+      data[:config][:bioc_check] = true
+      should include_sexp [:cmd, /.*BiocCheck.*/,
+                           assert: true, echo: true, timing: true]
+
```
Comment (jimhester) [inline spec/build/script/r_spec.rb]: This needs an end keyword

File: lib/travis/build/script/r.rb [hunk 3]
```diff
@@ -237,13 +237,17 @@ def script
...
-                sh.cmd 'Rscript -e "BiocCheck::BiocCheck(\"${PKG_TARBALL}\")"'
-            end
           end
           export_rcheck_dir
 
+          if config[:bioc_check]
+            # BiocCheck the package
+            sh.fold 'Bioc-check' do
+              sh.echo 'Checking with: BiocCheck( "${PKG_TARBALL}" ) '
+                  sh.cmd 'Rscript -e "BiocCheck::BiocCheck(\"${PKG_TARBALL}\")"'
```
Comment (jimhester) [inline lib/travis/build/script/r.rb]: There seems to be an indentation issue here

**Comment by LiNk-NY** [comment]
It looks like the build passes when `INTEGRATION_SPECS=1` otherwise it fails: 
https://travis-ci.org/travis-ci/travis-build/jobs/388454111
Thanks,
Marcel

**Comment by jimhester** [comment]
That looks like a spurious error to me, so I think your PR looks good, thanks!

**Comment by LiNk-NY** [comment]
Can we merge this PR and squash commits? Thanks all!
@jimhester @craigcitro @BanzaiMan

**Comment by LiNk-NY** [comment]
Hey all, @jimhester @craigcitro @joepvd @BanzaiMan
Any updates on merging this PR? It would also fix the issues that we are having with testing packages in `bioc-devel` which uses `R 3.5.0`.

**Comment by jimhester** [comment]
This LGTM I think it should be merged once it is reviewed by someone from travis.

**Comment by LiNk-NY** [comment]
@joepvd Jaaa, danke schön! :wink:

**Comment by LiNk-NY** [comment]
@joepvd @jimhester
This seems to be breaking Travis. Perhaps I missed something? 
Please see travis-ci/travis-yml#56 for an in-progress fix. 
Regards,
Marcel

**Pull request merged**